### PR TITLE
Plugins: Show latest commit date at plugin details right panel

### LINKED ISSUE #94247: Provide information on last commit date/time for plugin repository - should be stored rather than query GitHub API on page load

### PR DESCRIPTION
**Pull request by Ukochka**
**What is this feature?**
Show last commit date at plugin details right panel, which we get from gcom

**Why do we need this feature?**
We would like to show with this info if plugin is maintainable and up to date

**Which issue(s) does this PR fix?**:

Fixes #94247

### CODE REVIEW

File: public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx [hunk 1]
```diff
@@ -48,7 +48,21 @@ export function PluginDetailsRightPanel(props: Props): React.ReactElement | null
               <Text color="secondary">
                 <Trans i18nKey="plugins.details.labels.updatedAt">Last updated: </Trans>
               </Text>{' '}
-              <Text>{formatDate(new Date(plugin.updatedAt))}</Text>
+              <Text>{formatDate(new Date(plugin.updatedAt), { day: 'numeric', month: 'short', year: 'numeric' })}</Text>
```
Comment (academo) [inline public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx:51]: is this change intentional? it doesn't look like part of this PR scope.

File: public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx [hunk 2]
```diff
@@ -48,7 +48,21 @@ export function PluginDetailsRightPanel(props: Props): React.ReactElement | null
...
+            </Stack>
+          )}
+          {plugin?.details?.lastCommitDate && (
+            <Stack direction="column" gap={0.5}>
+              <Text color="secondary">
+                <Trans i18nKey="plugins.details.labels.lastCommitDate">Last commit date: </Trans>
```
Comment (academo) [inline public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx:57]: remove the space from inside the `<Trans>`

File: public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx (hunk 1 above)
Comment (Ukochka) [reply public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx:51]: Yes, it is, I decided that format 24 Oct 2024 is more readable than 24/10/2024, but this is just me.

**Review by academo — COMMENTED** [review]
The code looks good. Please make sure to add test cases:

* When the remote attribute comes and render
* When the remote attribute is undefined

**Review by academo — APPROVED** [review]
